### PR TITLE
Daos 4597 move daos_mgmt_set_params()out of mgmt module

### PR DESCRIPTION
**Pull request by GitHuaKuang**
Moved two apis from daso client mgmt to debug submodule:

    . Created mgmt/cli_debug.c and include/daos/sys_debug.h
    .renamed dc_mgmt_set_param to dc_debug_set_param
    .renamed daos_mgmt_set_params to daos_debug_set_params

    .renamed dc_mgmt_add_mark to dc_debug_add_mark
    .renamed daos_mgmt_add_mark to daos_debug_add_mark

   . Moved dc_debug_* to a new header file src/include/daos/sys_debug.h
Also:
   . removed dc_pool_extend

### CODE REVIEW

File: src/mgmt/cli_debug.c [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	struct mgmt_params_set_in	*in;
+	crt_endpoint_t			ep;
+	crt_rpc_t			*rpc = NULL;
+	crt_opcode_t			opc;
+	int				rc;
+
+	args = dc_task_get_args(task);
+	rc = dc_mgmt_sys_attach(args->grp, &cp_arg.sys);
+	if (rc != 0) {
+		D_ERROR("failed to attach to grp %s, rc "DF_RC".\n", args->grp,
```
Comment (ashleypittman) [inline src/mgmt/cli_debug.c:55]: No . on the end here after DF_RC.  Also applies elsewhere in this PR.
```suggestion
		D_ERROR("failed to attach to grp %s, rc "DF_RC"\n", args->grp,
```

**Review by ashleypittman — COMMENTED** [review]
PR title needs updating as well,

DAOS-4599 mgmt: Move two dc_mgmt apis to dc_debug

File: src/mgmt/cli_debug.c (hunk 1 above)
Comment (GitHuaKuang) [reply src/mgmt/cli_debug.c]: Done.

**Comment by GitHuaKuang** [comment]
The failures are reported to:
https://jira.hpdd.intel.com/browse/DAOS-6194

**Review by mjmac — APPROVED** [review]
LGTM

**Review by cibervicho — APPROVED** [review]

**Comment by GitHuaKuang** [comment]
@daos-stack/daos-gatekeeper can we land this one?

**Pull request merged**